templates/dashboard/subscription.php

```diff
@@ -1,3 +1,131 @@
+<?php
+
+if ( ! function_exists( 'wpuf_dashboard_get_subscription_data' ) ) {
+    function wpuf_dashboard_get_subscription_data( $user_sub ) {
+        global $wpdb;
+
+		$user_id = get_current_user_id();
+		$pack_id = $user_sub['pack_id'];
+		$subscription = wpuf()->subscription->get_subscription( $pack_id );
+
+        if ( ! $subscription || is_wp_error( $subscription ) || empty( $subscription->meta_value ) ) {
+            return [];
+        }
+		// Get payment gateway
+		$payment_gateway = $wpdb->get_var(
+            $wpdb->prepare(
+                "SELECT payment_type 
+            FROM {$wpdb->prefix}wpuf_transaction 
+            WHERE user_id = %d 
+            AND status = 'completed' 
+            ORDER BY created DESC",
+                $user_id
+            )
+		);
+		$payment_gateway = $payment_gateway ? strtolower( $payment_gateway ) : '';
+
+		// Get last payment date
+		$last_payment_date = $wpdb->get_var(
+            $wpdb->prepare(
+                "SELECT created 
+            FROM {$wpdb->prefix}wpuf_transaction 
+            WHERE user_id = %d 
+            AND status = 'completed' 
+            ORDER BY created DESC 
+            LIMIT 1",
+                $user_id
+            )
+		);
+
+		// Get billing cycle details
+		$cycle_number = intval( $subscription->meta_value['billing_cycle_number'] );
+		$cycle_period = $subscription->meta_value['cycle_period'];
+
+		// Get trial details
+		$trial_status = $subscription->meta_value['trial_status'];
+		$trial_duration = $subscription->meta_value['trial_duration'];
+		$trial_duration_type = $subscription->meta_value['trial_duration_type'];
+
+		return [
+			'payment_gateway' => $payment_gateway,
+			'last_payment_date' => $last_payment_date,
+			'cycle_number' => $cycle_number,
+			'cycle_period' => $cycle_period,
+			'trial_status' => $trial_status,
+			'trial_duration' => $trial_duration,
+			'trial_duration_type' => $trial_duration_type,
+		];
+	}
+}
+
+function display_subscription_details( $subscription_data ) {
+	$trial_html = get_trial_expiration_html( $subscription_data );
+	$billing_html = get_next_billing_html( $subscription_data );
+	?>
+        <br>
+	<?php if ( ! empty( $subscription_data['trial_status'] ) && 'on' === $subscription_data['trial_status'] ) : ?>
```

Comment:
please use `wpuf_is_checkbox_or_toggle_on()` function to check this type of checkbox, toggle values